Tests/NIOTestUtilsTests/NIOHTTP1TestServerTest.swift

```diff
@@ -0,0 +1,343 @@
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+import NIOTestUtils
+import XCTest
+
+class NIOHTTP1TestServerTest: XCTestCase {
+    private var group: EventLoopGroup!
+    private let allocator = ByteBufferAllocator()
+
+    override func setUp() {
+        self.group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    }
+
+    override func tearDown() {
+        XCTAssertNoThrow(try self.group.syncShutdownGracefully())
+        self.group = nil
+    }
+
+    func connect(serverPort: Int, responsePromise: EventLoopPromise<String>) throws -> EventLoopFuture<Channel> {
+        let bootstrap = ClientBootstrap(group: self.group)
+            .channelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
+            .channelInitializer { channel in
+                channel.pipeline.addHTTPClientHandlers(position: .first,
+                                                       leftOverBytesStrategy: .fireError).flatMap {
+                    channel.pipeline.addHandler(AggregateBodyHandler())
+                }.flatMap {
+                    channel.pipeline.addHandler(TestHTTPHandler(responsePromise: responsePromise))
+                }
+        }
+        return bootstrap.connect(host: "127.0.0.1", port: serverPort)
+    }
+
+    private func sendRequest(channel: Channel, uri: String, message: String) {
+        var requestBuffer = allocator.buffer(capacity: message.utf8.count)
+        requestBuffer.writeString(message)
+        var headers = HTTPHeaders()
+        headers.add(name: "Content-Type", value: "text/plain; charset=utf-8")
+        headers.add(name: "Content-Length", value: "\(requestBuffer.readableBytes)")
+
+        let requestHead = HTTPRequestHead(version: HTTPVersion(major: 1, minor: 1),
+                                          method: .GET,
+                                          uri: uri,
+                                          headers: headers)
+
+        channel.write(NIOAny(HTTPClientRequestPart.head(requestHead)), promise: nil)
+        channel.write(NIOAny(HTTPClientRequestPart.body(.byteBuffer(requestBuffer))), promise: nil)
+        channel.writeAndFlush(NIOAny(HTTPClientRequestPart.end(nil)), promise: nil)
+    }
+
+    func testTheExampleInTheDocs() {
+        // Setup the test environment.
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        let allocator = ByteBufferAllocator()
+        let testServer = NIOHTTP1TestServer(group: group)
+        defer {
+            XCTAssertNoThrow(try testServer.stop())
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+
+        // -- SNIP --
```

Comment:
What is this comment for?

Reply:
@Lukasa everything below the line serves as the example that's copied into the docs :). Just to have it exeutable.

Reply:
That was the intent, but on a second thought if we move `sendRequestTo` definition outside this function we won't need to snip anything.